crypto/s2n_hash.h

```diff
@@ -63,13 +66,34 @@ struct s2n_hash_state {
             MD5_CTX md5;
             SHA_CTX sha1;
         } md5_sha1;
-    } hash_ctx;
+    } low_level;
+    struct s2n_hash_evp_digest {
```

Comment:
I have re-worked the naming a bit in this most recent revision to address your comments. Let me know if this helps.